### ISSUE TITLE
Comments download have been limited to 50

### ISSUE THREAD

**Issue by Gabriel-Aragao** [instaloader/instaloader#2125]
Both command line and python module are downloading only 50 comments of each post, 

**To Reproduce**
instaloader --login=scrape.gcp.3 --comments --no-captions --no-pictures -- -SHORTCODE

**Expected behavior**
When a post had more  than 250 comments, it used to appear the 250 ... 500 ... 750.. and every comment downloaded.
Since yesterday it have been downloading olny 50 comments.

**Instaloader version**
4.10.1

**Comment by Gabriel-Aragao** [comment]
> Yes, I have exactly the same problem in recent days. It gives only 11 recent comments from each post!!!

Have you found a solution?

**Comment by jopapo** [comment]
Just to add something I noticed:

50 is graphql paging hard-coded client-side. So, it returns just the first page.

Somehow the backend is returning:
```
"page_info":{
      "has_next_page":false,
      "end_cursor":"None"
   },
```

If `has_next_page` is True, the lib tries to fetch more pages. 

https://github.com/instaloader/instaloader/blob/master/instaloader/nodeiterator.py#L140

Can it be something with the graphql lib limit request?

**Comment by saurabhtakle** [comment]
Does the old version return all comments?

**Comment by aandergr** [comment]
Indeed, this is reproducible. Thanks for reporting!

**Comment by ardian708** [comment]
Has anyone found the solution? this library should be fixed please

**Comment by aandergr** [comment]
It seems it is still quite reliable to query the comments of posts that have more than 50 comments via an iPhone endpoint.

For those in a hurry, I've put together a little gist for this before I built it into Instaloader: https://gist.github.com/aandergr/35ad3881e726e2be8925cf09e719a68b

**Comment by aandergr** [comment]
Fixed in [Version 4.10.3](https://github.com/instaloader/instaloader/releases/tag/v4.10.3).
